homeassistant/components/dreo/fan.py

```diff
@@ -0,0 +1,152 @@
+"""Support for Dreo fans."""
+from __future__ import annotations
+
+import logging
+import math
+from typing import Any
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage, int_states_in_range
+)
+from hscloud.const import FAN_DEVICE
+
+from . import MyConfigEntry
+from .entity import DreoEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+        hass: HomeAssistant,
+        config_entry: MyConfigEntry,
+        async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Fan from a config entry."""
+
+    async_add_entities([DreoFanHA(device, config_entry) for device in config_entry.runtime_data.fans])
+
+
+class DreoFanHA(DreoEntity, FanEntity):
+    """Dreo fan."""
+    _state = False
+    _mode = None
+    _speed = 0
+    _direction = None
+    _oscillate = False
+    _available = True
+
+    _attr_supported_features = (FanEntityFeature.PRESET_MODE
+                                | FanEntityFeature.SET_SPEED
+                                | FanEntityFeature.OSCILLATE)
+
+    def __init__(self, device, config_entry) -> None:
+        """Initialize the Dreo fan."""
+        super().__init__(device, config_entry)
+
+    @property
```

Comment:
`fan` determines `is_on` by the state of `percentage` and `preset`, it it necessary to change this or can you rely on existing functionality? 

https://github.com/home-assistant/core/blob/b80f7185b2e8e3a04727a59032a517cb87c9ce19/homeassistant/components/fan/__init__.py#L354-L359